src/python/WMCore/MicroService/MSPileup/DataStructs/MSPileupObj.py

```diff
@@ -138,6 +146,33 @@ def validate(self, pdict=None):
                 return False, msg
         return True, msg
 
+    def validateTransitionRecord(self, records, pdoc):
+        """
+        Validate given transition records
+
+        :param records: transition records
+        :param pdoct: pileup document
+        :return: boolean
+        """
+        for record in records:
+            keys = ['updateTime', 'customDID', 'DN', 'containerFraction']
+            for key in keys:
+                if key not in record:
+                    self.logger.error("key '%s' is not present in transition record %s", key, record)
+                    return False
+                val = record[key]
+                if key == 'updateTime' and val < pdoc['insertTime'] or val > time.time():
```

Comment:
I don't recall the operators precedence. So here is my question, is this line evaluated like:
```
if (key == 'updateTime' and val < pdoc['insertTime']) or val > time.time():
```
or
```
if key == 'updateTime' and (val < pdoc['insertTime'] or val > time.time()):
```
? 

Perhaps we should be explicit and define it like the latter.

Reply:
I made necessary change